File: gfapy/line/edge/containment/to_gfa2.py
import gfapy

class ToGFA2:

  @property
  def from_coords(self):
    """
    GFA2 positions of the alignment on the **from** segment.

    Returns
    -------
    (Integer|Lastpos, Integer|Lastpos)
    	begin and end

    Raises
    ------
    gfapy.RuntimeError
    	If the segment length cannot be determined, because the segment line is unknown.
    gfapy.ValueError
    	If the segment length is not specified in the segment line.
    """
    self._check_overlap()
    rpos = self.pos + self.overlap.length_on_reference()
    if rpos == self._lastpos_of("from_segment"):
      rpos = gfapy.LastPos(rpos)
    return [self.pos, rpos]

  @property
  def to_coords(self):
    """
    GFA2 positions of the alignment on the **to** segment
    """
    return [0, self._lastpos_of("to_segment")]

File: lib/rgfa/line/edge/containment/to_gfa2.rb
module RGFA::Line::Edge::Containment::ToGFA2

  # GFA2 positions of the alignment on the +from+ segment
  # @!macro [new] coords
  #   @return [(Integer|Lastpos,Integer|Lastpos)] begin and end
  #   @raise [RGFA::RuntimeError] if the segment length cannot be determined,
  #     because the segment line is unknown
  #   @raise [RGFA::ValueError] if the segment length is not specified
  #     in the segment line
  def from_coords
    check_overlap
    rpos = pos + overlap.length_on_reference
    rpos = rpos.to_lastpos if rpos == lastpos_of(:from)
    return [pos, rpos]
  end

  # GFA2 positions of the alignment on the +to+ segment
  # @!macro coords
  def to_coords
    return [0, lastpos_of(:to)]
  end

end
